lib/ohai/plugins/passwd.rb

```diff
@@ -42,6 +42,95 @@ def fix_encoding(str)
   end
 
   collect_data(:windows) do
-    # Etc returns nil on Windows
+    require "wmi-lite/wmi" unless defined?(WmiLite::Wmi)
+
+    unless etc
+      etc Mash.new
+
+      wmi = WmiLite::Wmi.new
+
+      etc[:passwd] = Mash.new
+      users = wmi.query("SELECT * FROM Win32_UserAccount Where LocalAccount = True")
+      users.each do |user|
+        uname = user["Name"].strip.downcase
+        Ohai::Log.debug("processing user #{uname}")
+        etc[:passwd][uname] = Mash.new
+        # apparently, chef/wmi-lite is too good to give us
```

Comment:
No offense, @adamedx  ;)

Reply:
:) That code was paged out of my memory -- if `wmi-lite` is giving you one of these [`wmi_instance`](https://github.com/chef/wmi-lite/blob/master/lib/wmi-lite/wmi_instance.rb) objects, maybe taking a look at that code will reveal a way to get at the properties. In theory the equivalent of an explicit `select` is more efficient and deterministic in the face of schema changes.
Also looking at `wmi-lite`, its built on top of `WIN32OLE`, so again there may be an implementation clue that might make it easier to use (or maybe hint at an improvement to `wmi-lite`).

At any rate, yeah, I think switching from shelling out to running in process seems like a huge perf gain, so I think the pain here is worth the tradeoff. Another option that lets you keep the PowerShell is to use a PowerShell session -- see [`New-PSSession`](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/new-pssession?view=powershell-7#example-1--create-a-session-on-the-local-computer). This would shell out once, but let you keep issuing commands to the same PowerShell -- you wouldn't pay repeated process startup cost, just the price of initial start and then marshalling input and output across processes.

There are probably other fun tricks such as finding a way to use .net, or some native stuff to call win32 apis, e.g. this fun from `chef`: https://github.com/chef/chef/blob/master/lib/chef/util/windows/net_user.rb#L34-L63. I guess you'd have to port it though since we're in `ohai` here...

Reply:
I am not familiar enough with `wmi-lite` to know if this is similar enough, but there's another plugin that iterates across the properties here: https://github.com/chef/ohai/blob/master/lib/ohai/plugins/windows/dmi.rb

Reply:
Thanks @phiggins ! I had tried similar things, but didn't make it work, but that's a great example. I will try again.